### PR TITLE
Allow 2D cubic interpolation

### PR DESCRIPTION
**Pull request by TomTranter**
# Description

Allow cubic interpolation of 2D data

Fixes # 2017

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: tests/unit/test_expression_tree/test_interpolant.py [hunk 1]
```diff
@@ -87,6 +87,11 @@ def test_interpolation_2_x_2d_y(self):
         np.testing.assert_array_almost_equal(
             interp.evaluate(y=np.array([0, 0])), 0, decimal=3
         )
+        # cubic
+        interp = pybamm.Interpolant(x, z, (var1, var2), interpolator="cubic")
+        np.testing.assert_array_almost_equal(
+            interp.evaluate(y=np.array([0, 0])), 0, decimal=3
+        )
```
Comment (tinosulzer) [inline tests/unit/test_expression_tree/test_interpolant.py:94]: I know you just copied the linear test but we really need a better test for this :P

**Review by tinosulzer — APPROVED** [review]
Looks good to me

File: tests/unit/test_expression_tree/test_interpolant.py (hunk 1 above)
Comment (TomTranter) [reply tests/unit/test_expression_tree/test_interpolant.py:94]: Yeah could check against z for a point in a finer interpolation

**Comment by rtimms** [comment]
Now in #2258

**Pull request closed without merging**